### PR TITLE
add failing xunit async fixture tests.

### PR DESCRIPTION
**Pull request by goblinfactory**
@SimonCropp  this is the failing test on it's own branch. 

So it's only the Fixture tests that cause a problem from what I can see/reproduce.

I have another branch I will submit a PR for separately in which I am unable to reproduce failing `async Task` tests, and am submitting that separately. I will comment on the current open tickets and ask if folk can update the tests until they fail. If not, we should close the other open issues. 

The extra tests will add some needed coverage.

### CODE REVIEW

File: ApprovalTests.Xunit2/CompatibilityWithXunit1AsyncFixtureTest.cs [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        private string _connectionString;
+
+        public CompatibilityWithXunit1AsyncFixutureTest(DummyTestStateClass state)
+        {
+            _connectionString = state.ConnectionString;
+        }
+
+        [Fact]
+        [UseReporter(typeof(DiffReporter))]
+        public async void Xunit2ShouldWorkWithAsyncFixtureTest()
```
Comment (SimonCropp) [inline ApprovalTests.Xunit2/CompatibilityWithXunit1AsyncFixtureTest.cs]: shouldnt this return a Task?

Comment (goblinfactory) [reply ApprovalTests.Xunit2/CompatibilityWithXunit1AsyncFixtureTest.cs]: I honestly don't know, I created that PR in February and I can't remember. Let me know if you need me to look at it if you have not fixed it already.

Comment (goblinfactory) [reply ApprovalTests.Xunit2/CompatibilityWithXunit1AsyncFixtureTest.cs]: just took a look, and yes, I can't see a reason why it should return async void. All I could remember was testing some weird edge cases, and thought perhaps there was some strange quirk that this was needed. Looking at the code it looks like a simple mistake. The important part of the test is ensuring (testing) that we succesfully  call `Approvals.Verify` from within an async block. The reason I wrote these in the first place (phew... feels like a lifetime ago) was because I  discovered I couldnt use approvals at a banking client for our async tests. It was the only way I could  report the approvals problem. I don't even have the sln open, I'm on another machine. If this test is failing the build, is it possible the problem still exists? i.e. it's not failing because it returns void (false negative), but failing because it's a positive negative. I say this because I saw that the PR request is failing a build and I have not even opened the project to look. So, yes, it should return `Task`.

**Comment by SimonCropp** [comment]
can you rebase on master?

**Comment by goblinfactory** [comment]
@SimonCropp  I'll take a look at rebasing tomorrow.

**Comment by goblinfactory** [comment]
@SimonCropp  mmm, I'm a little confused. 
You say you want me to "rebase from master", not sure why? I see you've merged these unit tests in to master already and all the tests for that merge passes : https://ci.appveyor.com/project/isidore/approvaltests-net/builds/28743137

File: ApprovalTests.Xunit2/CompatibilityWithXunit1AsyncFixtureTest.cs (hunk 1 above)
Comment (goblinfactory) [reply ApprovalTests.Xunit2/CompatibilityWithXunit1AsyncFixtureTest.cs]: yes it should.
